src/runtime/hexagon_remote/halide_host.cpp

```diff
@@ -116,19 +117,18 @@ __attribute__((weak)) void remote_register_buf(void* buf, int size, int fd);
 
 void halide_hexagon_host_malloc_init() {
     pthread_mutex_init(&allocations_mutex, NULL);
+    ion_fd = open("/dev/ion", O_RDONLY, 0);
```

Comment:
open/close should be ref counted, so there's no problem running this multiple times, as long as each open call has a matching close.

That said, the rest of this init/deinit code is not thread safe. It's not urgent to fix this because we know how these are called, but we should fix this at some point.